sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -812,8 +813,16 @@ EventHubAsyncClient buildAsyncClient() {
 
         final TracerProvider tracerProvider = new TracerProvider(ServiceLoader.load(Tracer.class));
 
+        AmqpClientOptions amqpClientOptions;
+        if (clientOptions != null && clientOptions instanceof AmqpClientOptions) {
+            amqpClientOptions = (AmqpClientOptions) clientOptions;
+        } else {
+            amqpClientOptions = new AmqpClientOptions();
+        }
+
         return new EventHubAsyncClient(processor, tracerProvider, messageSerializer, scheduler,
-            isSharedConnection.get(), this::onClientClose);
+            isSharedConnection.get(), this::onClientClose,
+            amqpClientOptions.getIdentifier());
```

Comment:
This can potentially be `null`. So, when `client.getIdentifier()` is called, it returns `null`. Is this what we want? Previously, it seemed like a UUID was generated when it was `null`.

Reply:
Fixed this. In latest code, builder will generate a identifier when call `buildAsyncClient()` if amqpClientOptions don't set a specific identifier.